scene/gui/rich_text_label.cpp

```diff
@@ -3412,6 +3412,21 @@ bool RichTextLabel::remove_paragraph(int p_paragraph, bool p_no_invalidate) {
 	selection.click_item = nullptr;
 	selection.active = false;
 
+	if (is_processing_internal()) {
+		set_process_internal(false);
```

Comment:
It's better to do this only when necessary. I'd change the logic to disable process after loop when there is no fx stack.

Reply:
Why? This will be less efficient, it will need to run the loop to the end to make sure there are no effects. Currently, it breaks on first effect. And if there were no effects, `is_processing_internal()` will return false, and it will skip the loop at all.

Reply:
Changing process is not trivial.
You can add `bool has_effect`, set it inside the loop and still break. Then only disable process when `has_effect` is false.